src/components/shapes/draw_newshape/attributes.js

```diff
@@ -78,7 +79,60 @@ module.exports = {
                 '*vertical* allows vertical extend.'
             ].join(' ')
         },
-
+        label: {
+            text: {
+                valType: 'string',
+                dflt: '',
+                editType: 'none',
+                description: 'Sets the text to display with shape.'
+            },
+            font: fontAttrs({
+                editType: 'none',
+                description: 'Sets the shape label text font.'
+            }),
+            textposition: {
+                valType: 'enumerated',
+                values: [
+                    'top left', 'top center', 'top right',
+                    'middle left', 'middle center', 'middle right',
+                    'bottom left', 'bottom center', 'bottom right',
+                    'start', 'middle', 'end',
+                ],
+                editType: 'none',
+                description: 'Sets the position of the label text relative to the shape.'
```

Comment:
Please improve `description` here to indicate which `values` are related to which line types.
Thank you!